apps/ui/src/pages/applications/index.tsx

```diff
@@ -58,26 +51,15 @@ const ApplicationViewer = () => {
 			if (data.state !== ApplicationStates.DRAFT && isEditMode) {
 				navigation(`/application/${data.id}/`, { replace: true });
 			}
-
-			setApplicationData(data);
-			setLoading(false);
-		} else if (data && data.isError) {
-			setLoading(false);
-
-			setError({
-				message: data.statusCode === 404 ? translate('errors.applicationNotFound.title') : data.message,
-				errors: data.statusCode === 404 ? translate('errors.applicationNotFound.message') : data.errors,
-			});
 		}
-	}, [data, isEditMode, navigation, translate]);
+	}, [data, isEditMode, navigation]);
 
-	// Should make a component for this
-	if (!applicationData || error || loading) return <ErrorPage error={error} loading={loading} />;
+	if (!data || isError || isLoading) return <ErrorPage loading={isLoading} error={error} />;
```

Comment:
`Error` and `Loading` are distinct states.

`ErrorPage` component is doing too much. It should handle just the error state and a different component can handle loading state. They're seperate components. Changing based on props is an anti pattern. 

In this case you might use props to change what type of error you're passing to display it differently.
`<ErrorPage error="FETCH_ERROR"/>`